src/sinks/datadog/metrics/service.rs

```diff
@@ -31,23 +31,6 @@ impl RetryLogic for DatadogMetricsRetryLogic {
     fn is_retriable_error(&self, error: &Self::Error) -> bool {
         error.is_retriable()
     }
-
-    fn should_retry_response(&self, response: &Self::Response) -> RetryAction {
```

Comment:
This is a dead code path because we call `DatadogApiError::from_result(result)` below.

Reply:
🤔  I'm missing how calling the `from_result` function makes this dead code path? The RetryLogic implementation is used in the service framework code ... Like, I see in `is_retriable_error()` , we are using the DatadogApiError. But in this function we aren't.

Reply:
`DatadogApiError::from_result(result)` converts `result` into an error unless the status code is 2xx. This means that `should_retry_response` will never be call on a `response` that is non-2xx. Rather, the non-2xx status codes are converted into errors and the retry logic is handled in `error.is_retriable()`.

Reply:
Ah, got it now. thanks for the added context. Would it be helpful maybe to add a comment here summarizing this? It seems like a not common thing we are doing for this sink.